docs/troubleshooting-guide/performance.md

```diff
@@ -0,0 +1,110 @@
+
+### The answer you want isn't cached.
+
+If you have just restarted Metabase, it  has to refresh all its caches, so performance on some questions will be slower the first time you view them. This only happens once per question, but if you have thousands of users, each with their own custom questions, you may see a lot of slowdowns occurring all at once.
+
+**How to detect this:** Look at Metabase's logs, or in the server's logs, to see when it was last restarted.
+
+**How to fix this:** Patience is a virtue.
+
+### The database is overloaded by other traffic.
+
+Metabase is usually not the only application using your database.
+
+**How to detect this:** Checking the performance logs of the database server or the machine it's running on often reveals that the real problem is caused by some third-party application.
+
+**How to fix this:** That depends on what those other applications are, how frequently they are making queries, whether the database can be replicated or the load can be moved onto other systems, and so on. If some of the applications involved are primarily being used for batch processing (e.g., daily or weekly reports), you can also check when those jobs are scheduled to ensure that they don't overlap.
+
+Note: you may also see your database being overloaded if you're using the same database as Metabase's app database and for your own data. We strongly recommend that you don't do this in a production system or if you have more than a handful of users.
+
+### The question itself is slow.
+
+Joining half a dozen tables, each with a few million rows, simply takes a lot of time. On the other hand, while we do our best to create fully-formed SQL queries from graphical questions, SQL snippets, and questions that use other questions as starting points, it's a hard problem---particularly across as many databases as we support. We also don't take advantage of every quirk of every backend database. For example, Redshift stores values in columns rather than rows: some queries that work well for row-oriented databases are slow on columns and vice versa.
+
+**How to detect this:**
+
+1. Run the same SQL that Metabase is running but using a different tool. For example, if you have created a question using the Notebook Editor and you are using Postgres as your database, you can view the SQL, copy it, and run it from the command line using `psql`. (If you have written the question in SQL you can just copy and paste it.) If the query runs noticeably faster this way than it does when you run it through Metabase, the problem is almost certainly one of the ones described above.
```

Comment:
I've seen multiple times, where someone was debugging performance and was running the query directly on the database vs through Metabase, which might have several network layers between.
Generally I recommend using DBeaver, since it's also JDBC based, which matches most of our drivers too, and then running the tool on the same location as Metabase (or SSH passthru), so it's easier to detect potential network bottlenecks.

Reply:
thanks - added